rest/client/sender.go

```diff
@@ -73,21 +82,13 @@ backgroundSender:
 		case <-ctx.Done():
 			return
 		case <-timer.C:
-			grip.CatchAlert(s.comm.SendLogMessages(ctx, s.logTaskData, buffer))
-			buffer = []apimodels.LogMessage{}
+			buffer = s.flush(ctx, buffer)
```

Comment:
I think it'd be better to set the buffer separately from flushing, just for clarity to the reader.